shell/platform/linux/public/flutter_linux/fl_binary_messenger.h

```diff
@@ -91,6 +93,14 @@ struct _FlBinaryMessengerInterface {
   GBytes* (*send_on_channel_finish)(FlBinaryMessenger* messenger,
                                     GAsyncResult* result,
                                     GError** error);
+
+  void (*resize_channel)(FlBinaryMessenger* messenger,
+                         const gchar* channel,
+                         int64_t new_size);
```

Comment:
Is int64_t the right type here? Should this be `size_t` or at least `uint64_t`, or is there a need to support a negative size?

I notice that the [Dart side of this](https://github.com/flutter/engine/blob/eba35482ccaaba7526a4c2a82383c8edc0e907a6/lib/ui/channel_buffers.dart#L78) doesn't seem to guard against negative sizes either, though the documentation seems to imply that capacity should always be >= 0.

Reply:
If we use size_t here the method will need to validate if it's greater than INT64_MAX. If we allow int64_t, then we let the server Dart side do the validation and return an error. I think in practice the user is unlikely to send a negative number and relying on the second method is probably less likely to cause an issue.

Reply:
Perhaps worth adding a debug-mode assertion in that case via FML_DCHECK